modules/standard/Map.chpl

```diff
@@ -242,10 +242,23 @@ module Map {
       return result;
     }
 
+    pragma "no doc"
+    proc const this(k: keyType) const
+    where isNonNilableClass(valType) {
+      _enter();
+      if !myKeys.contains(k) then
+        boundsCheckHalt("map index " + k:string + " out of bounds");
+      try! {
```

Comment:
This looks just like what I had in mind. Awesome!

Reply:
Is a return by `const` overload enough for non-nilable classes, or do we also want to provide an overload that returns a non-const value? I'm not sure if the difference matters, here.